src/theme/DocRoot/Layout/Sidebar/index.tsx

```diff
@@ -1,51 +1,74 @@
 import React, {useEffect, useState} from "react"
+import {useHistory} from "react-router-dom"
 import Sidebar from "@theme-original/DocRoot/Layout/Sidebar"
 import Search from "docusaurus-lunr-search/src/theme/SearchBar"
-import {useHistory} from "react-router-dom"
 import useIsBrowser from "@docusaurus/useIsBrowser"
+
 import PageSearchIcon from "@site/static/icons/basic/page-search.svg"
 import EnterKeyIcon from "@site/static/icons/basic/enter-key.svg"
 import UpDownKeyIcon from "@site/static/icons/basic/up-down-key.svg"
 import EscapeKeyIcon from "@site/static/icons/basic/escape-key.svg"
-import styles from "./Sidebar.module.css"
+import styles from "./styles.module.css"
 
-export default function SidebarWrapper(props) {
-  const [isSearchModalVisible, setIsSearchModalVisible] = useState(false)
+const CustomSearch = () => {
+  const [isSearchModalVisible, setIsSearchModalVisible] = useState<boolean>(false)
   const history = useHistory()
   const isBrowser = useIsBrowser()
-  const placeholder = isBrowser && window.navigator.platform.startsWith("Mac") ? "Search ⌘+K" : "Search Ctrl+K"
+  const placeholder = isBrowser
+    ? window.navigator.userAgent.startsWith("Mac")
+      ? "Search ⌘+K"
+      : "Search Ctrl+K"
+    : "Search"
 
+  // Function to handle opening the search modal
   function handleSearchClick() {
     setIsSearchModalVisible(true)
   }
 
+  // Function to handle closing the search modal
   function handleSearchModalClose() {
     setIsSearchModalVisible(false)
   }
 
+  // Function to control body scroll based on modal visibility
   function setBodyScroll() {
     document.body.style.overflow = isSearchModalVisible ? "hidden" : "auto"
   }
 
+  // Effect to handle body scroll based on modal visibility changes
   useEffect(() => {
     setBodyScroll()
     return () => {
       document.body.style.overflow = "initial"
     }
   }, [isSearchModalVisible])
 
-  function handleKeyPress(event) {
+  // Effect to focus on search input when modal becomes visible
+  useEffect(() => {
+    if (isSearchModalVisible) {
+      setTimeout(() => {
+        const searchInput = document.getElementById("search_input_react")
+        if (searchInput) {
+          searchInput.focus()
+        }
+      }, 50)
+    }
+  }, [isSearchModalVisible])
+
+  // Function to handle key press events
+  function handleKeyPress(event: KeyboardEvent) {
     if (event.key === "Escape") {
       handleSearchModalClose()
     }
     if (
-      (event.metaKey && event.key === "k" && navigator.platform.includes("Mac")) ||
-      (event.ctrlKey && event.key === "k" && navigator.platform.includes("Win"))
+      (event.metaKey && event.key === "k" && navigator.userAgent.includes("Mac")) ||
```

Comment:
this a unsafe api ref (https://developer.mozilla.org/en-US/docs/Web/API/Navigator/userAgent) consider using
https://github.com/bestiejs/platform.js

Reply:
using react-platform-js package for this now